config/routes.rb

```diff
@@ -1,6 +1,9 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  get 'posts/index'
+  get 'posts/show'
+root 'users#index'
+get 'users/:id', to: 'users#show', as: 'user'
+get 'users/:author_id/posts', to: 'posts#index', as: 'user_posts'
+get 'user/:author_id/posts/:id', to: 'posts#show', as: 'user_post'
```

Comment:
- [Optional] I think it looks clearer this:

```ruby
resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```

Reply:
Thanks for the suggestion.